code/modules/projectiles/guns/flamer.dm

```diff
@@ -307,6 +307,35 @@
 /obj/item/weapon/gun/flamer/som/mag_harness
 	starting_attachment_types = list(/obj/item/attachable/flamer_nozzle/wide, /obj/item/attachable/magnetic_harness)
 
+//dedicated engineer pyro kit flamer
+/obj/item/weapon/gun/flamer/big_flamer/marinestandard/engineer
+	name = "\improper FL-86 incinerator unit"
+	desc = "The FL-86 is a more light weight incinerator unit designed specifically to fit into its accompanying engineers bag. Can only be used with magazine fuel tanks however."
+	default_ammo_type = /obj/item/ammo_magazine/flamer_tank/large
+	allowed_ammo_types = list(
+		/obj/item/ammo_magazine/flamer_tank,
+		/obj/item/ammo_magazine/flamer_tank/large,
+		/obj/item/ammo_magazine/flamer_tank/large/X,
+	)
+	attachable_allowed = list(
+		/obj/item/attachable/flashlight,
+		/obj/item/attachable/magnetic_harness,
+		/obj/item/attachable/motiondetector,
+		/obj/item/attachable/buildasentry,
+		/obj/item/attachable/stock/t84stock,
+		/obj/item/attachable/flamer_nozzle,
+		/obj/item/attachable/flamer_nozzle/wide,
+		/obj/item/attachable/flamer_nozzle/long,
+	)
+	starting_attachment_types = list(/obj/item/attachable/flamer_nozzle, /obj/item/attachable/stock/t84stock)
+	var/obj/item/storage/holster/backholster/flamer/flamerpack
+
+/obj/item/weapon/gun/flamer/big_flamer/marinestandard/engineer/removed_from_inventory(mob/user)
+	. = ..()
+	flamerpack.handle_item_insertion(src)
```

Comment:
I am almost certain that this is not safe. It also seems pretty wack.

Reply:
safe meaning?

Reply:
so what would be safe